app/controllers/submissions_controller.rb

```diff
@@ -197,43 +202,48 @@ def collect_submissions
       head 400
       return
     end
-    assignment = Assignment.includes(:groupings).find(params[:assignment_id])
-    groupings = assignment.groupings.find(params[:groupings])
-    collectable = []
-    some_before_due = false
-    some_released = Grouping.joins(current_submission_used: :results)
-                            .where('results.released_to_students': true)
-                            .where(id: groupings)
-                            .pluck(:id).to_set
-    collection_dates = assignment.all_grouping_collection_dates
-    is_scanned_exam = assignment.scanned_exam?
-    groupings.each do |grouping|
-      unless is_scanned_exam
-        collect_now = collection_dates[grouping.id] <= Time.current
-        some_before_due = true unless collect_now
-        next unless collect_now
-      end
-      next if params[:override] != 'true' && grouping.is_collected?
-      next if some_released.include?(grouping.id)
+    begin
+      authorize! with: SubmissionPolicy
```

Comment:
For each authorization, please think carefully about how to report an unauthorized error message.
The main thing to consider is: will the user be able to see a button/link/etc. that will call this action? If yes, then a flash message might be appropriate. If no, then any attempt to access this route should give a 403 error.

This applies not just to this route but to every route that you are changing/will change as part of this PR.